pkg/metadata/metadata.go

```diff
@@ -0,0 +1,44 @@
+package metadata
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+)
+
+// metadata houses the metadata that will be written to the report directory after
+type metadata struct {
+	ClusterID      string `json:"cluster-id"`
+	ClusterName    string `json:"cluster-name"`
+	ClusterVersion string `json:"cluster-version"`
+	Environment    string `json:"environment"`
+	UpgradeVersion string `json:"upgrade-version,omitempty"`
+
+	// Metrics
```

Comment:
And maybe turn Metrics into map[string]string so we can do metadata.metrics.timetoclusteready

Reply:
I actually intentionally resisted making this a map so that the naming of things had a little bit more immediate oversight. I'd prefer to keep it this way if that's alright with you.

Reply:
kk